source/parameters/patch.md

```diff
@@ -2,5 +2,123 @@
 (s:patch.params)=
 ## PatchP:  Patch Parameters
 
-In Construction...
+```{figure} figures/patch.svg
+:name: f:patch
 
+A) A `Patch` element can align its exit face arbitrarily with respect to its entrance face. The
+red arrow illustrates a possible particle trajectory form entrance face to exit face. B) The
+reference length of a `Patch` element, if `ref_coords` is set to the default value of
+`exit_end`, is the longitudinal distance from the entrance origin to the exit origin using the
+reference coordinates at the exit end as shown. If `ref_coords` is set to `entrance_end`, the
+length of the patch will be equal to the `z_offset`.
+```
+
+The `PatchP` parameter group describes the shift the reference curve of a `Patch` or element.
+Also see [`ReferenceChangeP`](#s:ref.change.params) for the parameter group that can be used
+to change reference time and/or energy. 
+The components of `PatchP` are:
+```{code} yaml
+  x_offset          - Offset in x-direction.
+  y_offset          - Offset in y-direction.
+  z_offset          - Offset in z-direction.
+  x_rot             - Offset in x-direction.
+  y_rot             - Offset in x-direction.
+  z_rot             - Offset in x-direction.
+  flexible          - Default is False.
+                        True -> User sets offsets and rot. 
+                        False -> Offsets and rot from branch layout. 
+  ref_coords        - Coordinate system defining the length
+  user_sets_length  - Default is False. Is the element length User set? 
+```
+
+The transformation from `Patch` entrance coordinates to exit coordinates is given by [](#wws)
+with
+```{math}
+:label: lxyz
+
+\begin{align}
+  {\bf L} &= 
+    \begin{pmatrix} 
+      \text{x_offset} \\ \text{y_offset} \\ \text{z_offset} 
+    \end{pmatrix}
+    \\
+  {\bf S} &= {\bf R}_{y} (\text{y_rot}) \; {\bf R}_{x} (\text{x_rot}) \; {\bf R}_{z} (\text{z_rot}) 
+\end{align}
+```
+
+A straight line element like a `Drift` or a `Quadrupole` has the exit face parallel to the
+entrance face. With a `Patch` element, the entrance and exit faces can be arbitrarily oriented
+with respect to one another as shown in {numref}`f:patch`A.
+
+There are two different ways the orientation of the exit face is determined. Which way is used is
+determined by the setting of the `flexible` attribute.  With the `flexible` attribute set to
+`False`, the default, The exit face of the `Patch` will be determined from the offset, and rot
+parameters. This type of `Patch` is called
+"rigid" or "inflexible" since the geometry of the `Patch` is solely determined by the
+`Patch`'s attributes as set in the lattice file and is independent of everything else. Example:
+```{code} yaml
+pt: patch, z_offset = 3.2   ! Equivalent to a drift
```

Comment:
Probably also needs a little update to become a valid yaml snippet, e.g.,
```suggestion
pt: patch, z_offset = 3.2   # Equivalent to a drift
```
and a bit more

Reply:
Changed to:
```
- pt
    kind: Patch
    PatchP:
      z_offset: 3.2   # Equivalent to a drift.
```